### PR TITLE
Weird behaviour if back is pressed in 2nd onboarding (#775)

### LINKED ISSUE #775: Weird behaviour if back is pressed in 2nd onboarding
When creating a 2nd account, if I input and confirm the password, then press the back button **before** the console creates the account (before he sends the passphrase etc):

1. I'm taken back to the accounts screen, nothing unusual here.
2. Then, without further user action, after some seconds (presumably when the account is created) the app goes back to the console chat.

- This can only happen in the second account onboarding, because the first one doesn't show the back button. Somewhat related to #741.
- The fact that I'm not logged in, yet the app can "pull me back in" without my password, could be a hint of some kind of security issue?

### PR DESCRIPTION
**Pull request by alwx**
Fixes #775

### CODE REVIEW

File: src/status_im/chat/subs.cljs [hunk 1]
```diff
@@ -266,3 +268,13 @@
   (fn [_ [_ id]]
     (let [contacts (subscribe [:get :contacts])]
       (reaction (:photo-path (@contacts id))))))
+
+(register-sub :account-generation-in-progress?
```
Comment (rasom) [inline src/status_im/chat/subs.cljs]: Could we use some flag to figure that account generation is in progress? If not better to fetch the logic from the subscription into a separate function and use it in both subscription and handler.

(btw i'm not sure that it's fine to use `subscribe` inside `reaction` macro)

File: src/status_im/navigation/handlers.cljs [hunk 2]
```diff
@@ -50,6 +50,11 @@
   (fn [db [_ view-id]]
     (replace-view db view-id)))
 
+(defn- can-navigate-back? []
+  (and (not @(subscribe [:account-generation-in-progress?]))
```
Comment (rasom) [inline src/status_im/navigation/handlers.cljs]: So we definitely have this information in the `db` parameter that is passed to handler, it doesn't look a good idea to subscribe inside the handler

**Comment by alwx** [comment]
@rasom thank you for comments. I've simplified the code and removed the controversial `subscribe` inside `reaction`.

**Comment by rasom** [comment]
When I create a new account and click back button during generation I see then the next error
screenshot_2017-02-14-17-32-50

https://status-im.slack.com/files/roma/F45B20H1S/2017_02_14_17_28_02.mp4

**Comment by alwx** [comment]
This is not related to the issue.
Although I was unable to reproduce it, I've made one improvement that will possibly eliminate the bug.

**Comment by rasom** [comment]
This is related to inconsistency in the state that happens when a user presses back button at account generation, so I suppose that it is not the same issue as reported, but it is quite similar and, hence, is related.
Besides, I'm getting that error when i follow steps to reproduce of original issue, so..
It still can be reproduced on my phone.

Actually, we could hide back button at account generation, using the same flag. And this probably will fix both problems.

**Comment by alwx** [comment]
It won't help. In this case, it will be possible to go back by pressing "Back" hardware button on Android. 
It would be nice if you provide some logs.

**Comment by rasom** [comment]
We have a control over hardware back button https://github.com/status-im/status-react/blob/ace355515abc2d2809a50089890f4134b9f5df4c/src/status_im/android/core.cljs#L48-L48. Will provide logs soon.

**Pull request merged**